src/algebra/ordered_group.lean

```diff
@@ -1004,7 +1004,32 @@ sup_ind _ _ h1 h2
 end has_neg
 
 section add_group
-variables [add_group α] [linear_order α]
+variables [add_group α]
+
+section sub_neg
+
+@[simp] lemma with_top.neg_zero : (-0 : with_top α) = 0 :=
+by rw [←with_top.coe_zero, with_top.neg_coe, neg_zero]
+
+@[simp] lemma with_top.neg_neg (x : with_top α) : -(-x) = x :=
+begin
+  induction x using with_top.rec_top_coe;
+  simp [with_top.neg_coe]
+end
+
+lemma with_top.neg_eq_iff_neg_eq {x y : with_top α} :
```

Comment:
I think it's probably worth having `with_top.eq_neg_iff_eq_neg` too, just to match `eq_neg_iff_eq_neg`?